synapse/storage/util/id_generators.py

```diff
@@ -269,15 +280,17 @@ async def get_next(self):
         # Assert the fetched ID is actually greater than what we currently
         # believe the ID to be. If not, then the sequence and table have got
         # out of sync somehow.
-        assert self.get_current_token_for_writer(self._instance_name) < next_id
+        with self._lock:
+            assert self._current_positions.get(self._instance_name, 0) < next_id
 
         with self._lock:
             self._unfinished_ids.add(next_id)
 
         @contextlib.contextmanager
         def manager():
             try:
-                yield next_id
+                # Multiple by return factor so that we ID has correct sign.
```

Comment:
```suggestion
                # Multiply by the return factor so that the ID has correct sign.
```

Look, all the words were correctly spelled, you don't *also* expect me to use the *right* ones too, do you??